File: lib/languages/elixir/modules.ex
# File: math.ex
defmodule Math do
  # Public functions
  def add(a, b) do
    a + b
  end

  def subtract(a, b) do
    a - b
  end

  # Private function
  defp square(x) do
    x * x
  end
end

# File: main.ex
defmodule Main do
  def run do
    result = Math.add(5, 3)  # 8
    # Math.square(4)  # Error: square is private
  end
end

File: lib/languages/nim/modules.nim
# File: math.nim
# Public procedures are exported with *
proc add*(a, b: int): int =
  a + b

proc subtract*(a, b: int): int =
  a - b

# Private procedure (not exported)
proc square(x: int): int =
  x * x

# Alternatively, use explicit export statement
proc multiply(a, b: int): int =
  a * b

export multiply

# File: main.nim
import math

proc main() =
  let result = math.add(5, 3)  # 8
  # math.square(4)  # Error: square is private
  let product = math.multiply(4, 5)  # 20

when isMainModule:
  main()

